logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -146,7 +146,12 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
 
     @Override
     public synchronized void close() throws IOException {
-        this.lock.release();
+        if (this.lock != null){
+            this.lock.release();
+            if (this.lock.channel() != null && this.lock.channel().isOpen()) {
```

Comment:
@robbavey this is a little questionable imo, given that we are closing the `currentWriter` below as well.

I think we have `IOUtils` (the Commons IO one) on the classpath already (or something similar) and should probably `IOUtils.closeQuietly()` here (or otherwise suppress any possible exceptions on the channel close) to prevent the close actions below this one from failing.